numpy/core/tests/test_multiarray.py

```diff
@@ -5856,28 +5856,31 @@ def test_native_padding_2(self):
         self._check('^x3T{xi}', {'f0': (({'f0': ('i', 1)}, (3,)), 1)})
 
     def test_trailing_padding(self):
-        # Trailing padding should be included, *and*, the item size
-        # should match the alignment if in aligned mode
+        # Trailing padding should not be added according to struct spec.
         align = np.dtype('i').alignment
         size = np.dtype('i').itemsize
 
-        def aligned(n):
-            return align*(1 + (n-1)//align)
-
         base = dict(formats=['i'], names=['f0'])
 
-        self._check('ix',    dict(itemsize=aligned(size + 1), **base))
-        self._check('ixx',   dict(itemsize=aligned(size + 2), **base))
-        self._check('ixxx',  dict(itemsize=aligned(size + 3), **base))
-        self._check('ixxxx', dict(itemsize=aligned(size + 4), **base))
-        self._check('i7x',   dict(itemsize=aligned(size + 7), **base))
```

Comment:
The PEP3118  spec is unclear about this.

One could argue the `struct` module has set no precedent in judging how alignment and padding work here since it doesn't implement the `T{}` format. We might then feel free to set the precedent here in this PR, deciding that aligned formats add trailing padding only inside `T{}`.